File: ballonsmenu/src/main/java/com/karsu/ballonsmenu/Piece/Dot.java
package com.karsu.ballonsmenu.Piece;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.graphics.drawable.GradientDrawable;

import com.karsu.ballonsmenu.R;
import com.karsu.ballonsmenu.Util;

/**
 * Created by Erkan Kaplan on 2026-02-03
 */

final class Dot extends BoomPiece {

    public Dot(Context context) {
        super(context);
    }

    @Override
    public void init(int color, float cornerRadius) {
        Drawable backgroundDrawable;
        if (cornerRadius < 0) backgroundDrawable = Util.getDrawable(this, R.drawable.piece_dot, null)
                .mutate();
        else backgroundDrawable = Util.getDrawable(this, R.drawable.piece, null)
                .mutate();
        ((GradientDrawable)backgroundDrawable).setColor(color);
        if (cornerRadius >= 0) ((GradientDrawable)backgroundDrawable).setCornerRadius(cornerRadius);
        Util.setDrawable(this, backgroundDrawable);
    }

    @Override
    public void setColor(int color) {
        ((GradientDrawable)getBackground()).setColor(color);
    }

    @Override
    public void setColorRes(int colorRes) { setColor(Util.getColor(getContext(), colorRes)); }
}

File: ballonsmenu/src/main/kotlin/com/karsu/ballonsmenu/piece/Dot.kt
/** Created by Erkan Kaplan on 2026-02-03 */
package com.karsu.ballonsmenu.piece

import android.content.Context
import android.graphics.drawable.GradientDrawable
import com.karsu.ballonsmenu.R
import com.karsu.ballonsmenu.Util

internal class Dot(context: Context) : KarSuPiece(context) {

    override fun init(color: Int, cornerRadius: Float) {
        val backgroundDrawable = if (cornerRadius < 0) {
            Util.getDrawable(this, R.drawable.piece_dot, null).mutate()
        } else {
            Util.getDrawable(this, R.drawable.piece, null).mutate()
        }

        (backgroundDrawable as GradientDrawable).setColor(color)
        if (cornerRadius >= 0) {
            backgroundDrawable.cornerRadius = cornerRadius
        }
        Util.setDrawable(this, backgroundDrawable)
    }

    override fun setColor(color: Int) {
        (background as? GradientDrawable)?.setColor(color)
    }

    override fun setColorRes(colorRes: Int) {
        setColor(Util.getColor(context, colorRes))
    }
}
